### DISCUSSION TITLE
[Enhancement] Native Icons

### DISCUSSION THREAD

**Discussion by mhrastegari** [dotnet/maui#72] [Ideas]
## Summary

Having native icons could be a helpful feature and saves our time for searching, adding and thinking about which icon pack is suts for all platforms.

## Example

I think you can do something like the UWP's Xaml :

![Screenshot (725)](https://user-images.githubusercontent.com/42671084/82553748-e5626380-9b79-11ea-9b5f-d4a3ae49433f.png)

 ## Platforms

UWP/WPF : Segoe mdl2 assets
Android : Material icons
iOS, macOS : sf symbols

**Comment by Martinedo** [comment]
If there is a icon set per platform, then a property to persist same or platform specific icon would be nice. For example: 
`IconSet`:

- `Default`- Icon is platform specific
- `Android` - On all platforms the Android icon is shown
- `iOS` - On all platforms the iOS icon is shown
- `UWP` - On all platforms the UWP icon is shown
- etc

**Comment by mhrastegari** [comment]
Yeah! it's a really cool idea @Martinedo

**Comment by yoshiask** [comment]
@Martinedo That's a great idea, but how would you do that? The icon sets for each platform are copyrighted, and apps are only licensed to use them on their platforms (for example, it's against Microsoft's license to use Segoe MDL2 Assets in anything that's not running on a Microsoft platform).

**Reply by mmarczell-graphisoft** [reply]
Maui could decide to only support Apple icons on Apple platforms, etc.

**Comment by Martinedo** [comment]
@yoshiask yes, you are right. That's something I didn't think through. My suggestion can only work for assets without a restricting licence.

**Comment by yoshiask** [comment]
This proposal might be doable now that Microsoft has released the Fluent System Icons, which are designed for use in cross-platform mobile apps. I already wrote a library (for UWP) which, in theory, could be modified to work with other platforms, including Xamarin.Forms and MAUI.
